File: api/app/model/thread.go
package model

import (
	"time"
)

type ThreadID uint64

type Thread struct {
	// ID ...
	ID ThreadID
	// UserID ...
	UserID UserID
	// RoomID ...
	RoomID RoomID
	// Text ...
	Text string
	// CreatedAt ...
	CreatedAt time.Time
	// UpdatedAt ...
	UpdatedAt time.Time
	// Img 画像データ
	Img *string
}

type ThreadUser struct {
	Thread Thread
	User   ShowUser
}

type IndexThread struct {
	ThreadUser   ThreadUser
	CountContent *int
}

File: client/lib/model/thread.ts
import { User } from "./user";

export class Thread {
    constructor(
        public id: number,
        public roomID: number,
        public userID: number,
        public text: string,
        public createdAt: Date,
        public updatedAt: Date,
        public img?: string,
    ) {}
}

export interface ThreadUser {
    thread: Thread;
    user: Omit<User, "email" | "password">;
}

export interface IndexThread {
    threadUser: ThreadUser;
    countContent: number;
}

export interface AllThread {
    indexThreads: IndexThread[];
    nextID: number | null;
}
